src/Application.vala

```diff
@@ -230,32 +231,43 @@ public class Terminal.Application : Gtk.Application {
         var window = (MainWindow) active_window;
         bool new_window;
 
-        if (window == null || options.lookup ("new-window", "b", out new_window) && new_window) {
+        if (window == null || (options.lookup ("new-window", "b", out new_window) && new_window)) {
             /* Uncertain whether tabs should be restored when app is launched with working directory from commandline.
              * Currently they are set to restore (subject to the restore-tabs setting).
              * If it is desired that tabs should never be restored in these circimstances add another check below.
              */
-            bool restore_tabs = !("commandline" in options || "execute" in options) || window == null;
-            window = new MainWindow (this, restore_tabs);
+            bool restore_tabs = !("commandline" in options || "execute" in options || "working-directory" in options) || window == null;
+            window = new MainWindow (this, restore_tabs, false);
         }
 
-        unowned var working_directory = command_line.get_cwd ();
         unowned string[] commands;
         unowned string command;
+        unowned string wd;
```

Comment:
keep using `command_line.get_cwd()` it's already set to be the directory passed in "working-directory" or the directory of the current invocation.